extensions/Worker.Extensions.ServiceBus/src/ServiceBusReceivedMessageConverter.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Primitives;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+internal class ServiceBusReceivedMessageConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        if (context is null)
+        {
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+        return context.Source switch
+        {
+            ModelBindingData binding => ConvertToServiceBusReceivedMessage(context, binding),
+            _ => new ValueTask<ConversionResult>(ConversionResult.Unhandled()),
```

Comment:
Are there no collection use cases with service bus? I thought this:

`[ServiceBusTrigger("<queue_name>", Connection = "<connection_name>")] ServiceBusReceivedMessage[] messages`

was a valid scenario, so I would expect `CollectionModelBindingData` to be handled here

Reply:
Good catch - yes we should handle collections here.

Reply:
Question - do we need to do anything special on the extension side or will this work automatically for collections based on this - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusExtensionConfigProvider.cs#L100

Reply:
For Blob, we did [collection handling](https://github.com/Azure/azure-sdk-for-net/blob/abf92e3b4d559ac1cee311a274b7a3e3f662248c/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs#L110-L111) on extension side.